enclave-runtime/src/rpc/worker_api_direct.rs

```diff
@@ -192,6 +230,38 @@ fn execute_getter_inner<G: ExecuteGetter>(
 	Ok(getter_result)
 }
 
+fn forward_dcap_quote_inner(params: Params) -> Result<OpaqueExtrinsic, String> {
+	let hex_encoded_params = params.parse::<Vec<String>>().map_err(|e| format!("{:?}", e))?;
+
+	let request =
+		Request::from_hex(&hex_encoded_params[0].clone()).map_err(|e| format!("{:?}", e))?;
+
+	let encoded_quote_to_forward: Vec<u8> = request.cyphertext;
+
+	let url = String::new();
+	let ext = generate_dcap_ra_extrinsic_from_quote_internal(url, &encoded_quote_to_forward)
+		.map_err(|e| format!("{:?}", e))?;
+
+	Ok(ext)
+}
+
+fn attesteer_forward_ias_attestation_report_inner(
+	params: Params,
+) -> Result<OpaqueExtrinsic, String> {
+	let hex_encoded_params = params.parse::<Vec<String>>().map_err(|e| format!("{:?}", e))?;
+
+	let request =
+		Request::from_hex(&hex_encoded_params[0].clone()).map_err(|e| format!("{:?}", e))?;
+
+	let ias_attestation_report: Vec<u8> = request.cyphertext;
+
+	let url = String::new();
+	let ext = generate_ias_ra_extrinsic_from_der_cert_internal(url, &ias_attestation_report)
+		.map_err(|e| format!("{:?}", e))?;
+
+	Ok(ext)
+}
```

Comment:
I think you messed up some stuff here:

1. The request will be decoded into the wrong thing because the first few bytes are interpreted as the `ShardIdentifier`, which you don't pass in the request (which is fine). So the hex encoded params does only contain the hex encoded remote attestation. Hence, what you need to do is simply:

```rust
// also check that `hex_encoded_params.length == 1`
let ias_attestation_report = itp_utils::hex::decode_hex(hex_encoded_params[0]).map_err(...);
```

Where are you sending the remote attestation to the chain? You just send back the ready to send extrinsic to the client. The attesteer should send it to the chain, though. Is this a programmatic error, or do we have a misunderstanding about what the attesteer should do?

Reply:
It was the former, I fixed it accordingly in https://github.com/integritee-network/worker/pull/1201/commits/c92056114ac74e998616387c62196b1773bd9241.